packages/lib/holidays/GoogleHolidayService.ts

```diff
@@ -0,0 +1,216 @@
+    const cacheDays = getHolidayCacheDays();
+    const staleDate = dayjs().subtract(cacheDays, "days").toDate();
+
+    const cachedEntry = await prisma.holidayCache.findFirst({
+      where: {
+        countryCode,
+        year,
+      },
+      orderBy: {
+        updatedAt: "desc",
+      },
+    });
+
+    if (!cachedEntry) {
+      return true;
+    }
+
+    return cachedEntry.updatedAt < staleDate;
+  }
+
+  async getHolidaysForCountry(countryCode: string, year: number): Promise<CachedHoliday[]> {
+    const isStale = await this.isCacheStale(countryCode, year);
+
+    if (isStale) {
+      await this.refreshCache(countryCode, year);
+    }
+
+    const cached = await prisma.holidayCache.findMany({
+      where: {
+        countryCode,
+        year,
+      },
+      orderBy: {
+        date: "asc",
+      },
+    });
+
+    return cached.map((h) => ({
+      id: h.id,
+      countryCode: h.countryCode,
+      eventId: h.eventId,
+      name: h.name,
+      date: h.date,
+      year: h.year,
+    }));
+  }
+
+  async refreshCache(countryCode: string, year: number): Promise<void> {
+    const calendarConfig = GOOGLE_HOLIDAY_CALENDARS[countryCode];
+    if (!calendarConfig) {
+      return;
+    }
+
+    try {
+      const holidays = await this.fetchHolidaysFromGoogle(countryCode, year);
+
+      await prisma.$transaction(async (tx) => {
+        await tx.holidayCache.deleteMany({
+          where: {
+            countryCode,
+            year,
+          },
+        });
+
+        if (holidays.length > 0) {
+          await tx.holidayCache.createMany({
+            data: holidays.map((h) => ({
+              countryCode: h.countryCode,
+              calendarId: calendarConfig.calendarId,
+              eventId: h.eventId,
+              name: h.name,
+              date: h.date,
+              year: h.year,
+            })),
+            skipDuplicates: true,
+          });
+        }
+      });
+    } catch (error) {
+      console.error(`Failed to refresh holiday cache for ${countryCode}:`, error);
+    }
+  }
+
+  async getHolidaysInRange(countryCode: string, startDate: Date, endDate: Date): Promise<CachedHoliday[]> {
+    const startYear = dayjs(startDate).year();
+    const endYear = dayjs(endDate).year();
+
+    for (let year = startYear; year <= endYear; year++) {
+      const isStale = await this.isCacheStale(countryCode, year);
+      if (isStale) {
+        await this.refreshCache(countryCode, year);
+      }
+    }
+
+    const cached = await prisma.holidayCache.findMany({
+      where: {
+        countryCode,
+        date: {
+          gte: startDate,
+          lte: endDate,
+        },
+      },
+      orderBy: {
+        date: "asc",
+      },
+    });
+
+    return cached.map((h) => ({
+      id: h.id,
+      countryCode: h.countryCode,
+      eventId: h.eventId,
+      name: h.name,
+      date: h.date,
+      year: h.year,
+    }));
+  }
+}
+
+export const GoogleHolidayService = new GoogleHolidayServiceClass();
```

Comment:
why not just exporting the class and let the caller instantiate it? You can refer to other examples of services, for example, `packages/lib/server/service/ApiKeyService.ts`